cider/tests/nextgen/operator/FilterProjectTest.cpp

```diff
@@ -100,6 +108,82 @@ TEST_F(FilterProjectTests, BasicTest) {
   executeFilterTest<JITTypeTag::INT32>(6, 6, builder);
 }
 
+TEST_F(FilterProjectTests, IsNullTest) {
+  LLVMJITModule module("TestIsNull", true);
+
+  auto builder = [](JITFunction* function) {
+    auto col_var0 =
+        std::make_shared<Analyzer::ColumnVar>(SQLTypeInfo(SQLTypes::kINT), 100, 0, 0);
+    auto col_var1 =
+        std::make_shared<Analyzer::ColumnVar>(SQLTypeInfo(SQLTypes::kINT), 100, 1, 0);
+    auto col_var2 =
+        std::make_shared<Analyzer::ColumnVar>(SQLTypeInfo(SQLTypes::kDOUBLE), 100, 2, 0);
+
+    // constant
+    auto const_var = makeConstant(5);
+
+    // col_0 IS NULL
+    auto cmp_expr =
+        std::make_shared<Analyzer::UOper>(SQLTypes::kBOOLEAN, SQLOps::kISNULL, col_var0);
+
+    // col_1 + 5
+    auto add_expr = std::make_shared<Analyzer::BinOper>(
+        SQLTypes::kINT, SQLOps::kPLUS, SQLQualifier::kONE, col_var1, const_var);
+
+    // select col_1 + 5 from test where col_0 IS NULL
+    auto source = ArrowSourceTranslator(
+        std::vector<ExprPtr>{col_var0, col_var1, col_var2},
+        std::make_unique<ColumnToRowTranslator>(
+            std::vector<ExprPtr>{col_var0, col_var1, col_var2},
+            std::make_unique<FilterTranslator>(
+                cmp_expr,
+                std::make_unique<ProjectTranslator>(
+                    add_expr, std::make_unique<MockRowToColumnTranslator>(1)))));
+
+    Context context(function);
+    source.consume(context);
+
+    function->createReturn();
+  };
+  JITFunctionPointer func =
+      JITFunctionBuilder()
+          .setFuncName("test_C2R_translator")
+          .registerModule(module)
+          .addReturn(JITTypeTag::VOID)
+          .addParameter(JITTypeTag::POINTER, "in", JITTypeTag::INT8)
+          .addParameter(JITTypeTag::POINTER, "out", JITTypeTag::INT32)
+          .addProcedureBuilder(builder)
+          .build();
+
+  module.finish();
+
+  std::vector<int> vec0{1, 3, 2, 1, 6};
+  std::vector<int> vec1{1, 2, 3, 4, 5};
```

Comment:
vec1 and vec2 are not really in use?

Reply:
`vec1` is used while `vec2` is not. But it's okay to put them here to cover input arrowarray which has extra columns.